proxy/go/golang.go

```diff
@@ -24,20 +30,20 @@ func IsValid(path string) bool {
 }
 
 // Factory returns an instance of the Proxy.
-func Factory() *Proxy {
-	return &Proxy{}
+func Factory() proxy.Proxy {
+	return &golang{}
 }
 
-// Proxy is a project proxy for a Gradle-based Java project.
-type Proxy struct{}
+// golang is a project proxy for a Gradle-based Java project.
+type golang struct{}
```

Comment:
I renamed the implementation struct to something language specific. This could probably be better named.